### PR TITLE
Add a new net6.0 TargetFramework

### PR DESCRIPTION
**Pull request by Anipik**
- adding a new target framework net6.0
- making coreclr tool target net5.0
- make packages containing net5.0 exclude netcoreappcurrent

### CODE REVIEW

File: src/libraries/shims/ApiCompat.proj [hunk 1]
```diff
@@ -111,7 +111,7 @@
       <_previousNetCoreAppBaselineParam Condition="'$(UpdatePreviousNetCoreAppBaseline)' == 'true'">&gt; &quot;$(_previousNetCoreAppBaselineFile)&quot;</_previousNetCoreAppBaselineParam>
     </PropertyGroup>
  
-    <Error Condition="'$(NetCoreAppCurrent)' != 'net5.0'" Text="Update value of PreviousNetCoreApp" />
+    <Error Condition="'$(NetCoreAppCurrent)' != 'net6.0'" Text="Update value of PreviousNetCoreApp" />
```
Comment (ericstj) [inline src/libraries/shims/ApiCompat.proj:114]: You need to update the values
https://github.com/dotnet/runtime/blob/7cbc26c13fd41d7fa879bdcbe44f0a4513f24be0/src/libraries/shims/ApiCompat.proj#L8-L10

To test against netcoreapp5.0 API.

File: eng/packaging.targets [hunk 2]
```diff
@@ -38,4 +38,11 @@
     <MSBuild Projects="$(PkgProjPath)"
              Targets="Build" />
   </Target>
+
+  <ItemGroup>
+    <DefaultValidateFramework Include="$(NetCoreAppCurrent)">
```
Comment (ericstj) [inline eng/packaging.targets:43]: We should make this change in arcade.  OK to do so here for now, but please open an issue to track making it in arcade.

File: Directory.Build.targets [hunk 3]
```diff
@@ -22,4 +22,20 @@
     <NETCoreAppMaximumVersion>$(MajorVersion).$(MinorVersion)</NETCoreAppMaximumVersion>
   </PropertyGroup>
 
+  <Target Name="AddNetCoreAppCurrentToKnownFrameworkReference"
```
Comment (ericstj) [inline Directory.Build.targets:25]: Does this have to be a target?  Can it be just an ItemGroup?

File: src/coreclr/src/tools/Common/JitInterface/ThunkGenerator/ThunkGenerator.csproj [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>$(NetCoreAppCurrent)</TargetFramework>
+    <TargetFramework>net5.0</TargetFramework>
```
Comment (ericstj) [inline src/coreclr/src/tools/Common/JitInterface/ThunkGenerator/ThunkGenerator.csproj:5]: Why are all these tools staying on 5.0?  Is it because they need to run on the current tool framework and that's still 5.0?

File: src/libraries/shims/ApiCompat.proj (hunk 1 above)
Comment (Anipik) [reply src/libraries/shims/ApiCompat.proj:114]: I was waiting to until we release the stable 5.0 stable bits.

File: eng/packaging.targets (hunk 2 above)
Comment (Anipik) [reply eng/packaging.targets:43]: Yeah thats the plan :P

File: src/libraries/shims/ApiCompat.proj (hunk 1 above)
Comment (ericstj) [reply src/libraries/shims/ApiCompat.proj:114]: We don't need stable bits.  We can switch to RC and file an issue to update it.  Technically we could change this is soon as be branch for release and move to v.Next.  It protects us from breaking API that was added in the last release.

File: eng/packaging.targets (hunk 2 above)
Comment (ericstj) [reply eng/packaging.targets:43]: Ok, please file issue and add a comment

File: Directory.Build.targets (hunk 3 above)
Comment (Anipik) [reply Directory.Build.targets:25]: We dont have value of ```BundledNETCoreAppPackageVersion``` at evaluation time. It is generated on the fly.

File: src/libraries/pkg/test/packageTest.targets [hunk 5]
```diff
@@ -3,7 +3,7 @@
   <PropertyGroup>
     <_targetFrameworkVersionIndex>$(TargetFramework.IndexOfAny(".-0123456789"))</_targetFrameworkVersionIndex>
     <_targetFrameworkIdentifier Condition="'$(_runtimeOSVersionIndex)' != '-1'">$(TargetFramework.SubString(0, $(_targetFrameworkVersionIndex)))</_targetFrameworkIdentifier>
-    <!-- For net5.0 and later use netcoreapp folder. -->
+    <!-- For net6.0 and later use netcoreapp folder. -->
```
Comment (ViktorHofer) [inline src/libraries/pkg/test/packageTest.targets:6]: This comment change should be reverted

File: src/coreclr/tests/scripts/run-corefx-tests.py [hunk 6]
```diff
@@ -297,7 +297,7 @@ def main(args):
 
     # Gather up some arguments to pass to the different build scripts.
 
-    common_config_args = '-configuration Release -framework net5.0 -os %s -arch %s' % (clr_os, arch)
+    common_config_args = '-configuration Release -framework net6.0 -os %s -arch %s' % (clr_os, arch)
```
Comment (ViktorHofer) [inline src/coreclr/tests/scripts/run-corefx-tests.py:300]: Who uses that script? cc @trylek

File: src/coreclr/src/tools/Common/JitInterface/ThunkGenerator/ThunkGenerator.csproj (hunk 4 above)
Comment (ViktorHofer) [reply src/coreclr/src/tools/Common/JitInterface/ThunkGenerator/ThunkGenerator.csproj:5]: Is it because VS doesn't support net6.0 yet?

File: eng/common/performance/performance-setup.ps1 [hunk 7]
```diff
@@ -36,7 +36,7 @@ $HelixSourcePrefix = "pr"
 $Queue = "Windows.10.Amd64.ClientRS4.DevEx.15.8.Open"
 
 # TODO: Implement a better logic to determine if Framework is .NET Core or >= .NET 5.
-if ($Framework.StartsWith("netcoreapp") -or ($Framework -eq "net5.0")) {
+if ($Framework.StartsWith("netcoreapp") -or ($Framework -eq "net6.0")) {
```
Comment (ViktorHofer) [inline eng/common/performance/performance-setup.ps1:39]: Changes in eng/common should be made in Arcade.

File: eng/common/performance/performance-setup.sh [hunk 8]
```diff
@@ -4,7 +4,7 @@ source_directory=$BUILD_SOURCESDIRECTORY
 core_root_directory=
 baseline_core_root_directory=
 architecture=x64
-framework=net5.0
+framework=net6.0
```
Comment (ViktorHofer) [inline eng/common/performance/performance-setup.sh:7]: Same about eng/common

File: Directory.Build.targets [hunk 9]
```diff
@@ -22,4 +22,16 @@
     <NETCoreAppMaximumVersion>$(MajorVersion).$(MinorVersion)</NETCoreAppMaximumVersion>
   </PropertyGroup>
 
+  <ItemGroup>
+    <KnownFrameworkReference Include="Microsoft.NETCore.App"
```
Comment (ViktorHofer) [inline Directory.Build.targets:26]: Add comment about why this is needed and open an issue in dotnet/sdk to discuss timeliness?

**Comment by ViktorHofer** [comment]
Can you please verify if vstest still works? I believe the xunit.visualstudio package and vstest were hardcoding on checking for "net5.0" of "netcoreapp". cc @nohwnd

File: src/coreclr/src/tools/Common/JitInterface/ThunkGenerator/ThunkGenerator.csproj (hunk 4 above)
Comment (tannergooding) [reply src/coreclr/src/tools/Common/JitInterface/ThunkGenerator/ThunkGenerator.csproj:5]: Would it be better to define a `$(NetCoreToolCurrent)` or similar, so its a central replace for tools vs libraries, both now and in the future?

**Comment by nohwnd** [comment]
@ViktorHofer yes, it should work. The change that caused that problem was rolled back somewhere between Preview8 and 5.0.100-**rc.1.20451.10**.

File: src/coreclr/src/tools/r2rtest/Commands/CompileNugetCommand.cs [hunk 10]
```diff
@@ -80,7 +80,7 @@ public static int CompileNuget(BuildOptions options)
                     }
 
                     // This is not a reliable way of building the publish folder
-                    string publishFolder = Path.Combine(appFolder, @"artifacts\Debug\net5.0\publish");
+                    string publishFolder = Path.Combine(appFolder, @"artifacts\Debug\net6.0\publish");
```
Comment (ViktorHofer) [inline src/coreclr/src/tools/r2rtest/Commands/CompileNugetCommand.cs:83]: Instead of changing this over and over again, can we just pass the moniker in as a constant?

File: src/installer/tests/Assets/TestProjects/HammerServiceApp/Location/Location.csproj [hunk 11]
```diff
@@ -1,7 +1,8 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>net5.0</TargetFramework>
+    <TargetFramework>net6.0</TargetFramework>
```
Comment (ViktorHofer) [inline src/installer/tests/Assets/TestProjects/HammerServiceApp/Location/Location.csproj:4]:
```suggestion
    <TargetFramework>$(NetCoreAppCurrent)</TargetFramework>
```

File: src/installer/tests/Assets/TestProjects/TestWindowsOsShimsApp/TestWindowsOsShimsApp.csproj [hunk 12]
```diff
@@ -7,6 +7,7 @@
     <RuntimeFrameworkVersion>$(MNAVersion)</RuntimeFrameworkVersion>
     <DefineConstants Condition="'$(OS)' == 'Windows_NT'">WINDOWS;$(DefineConstants)</DefineConstants>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <NETCoreAppMaximumVersion>6.0</NETCoreAppMaximumVersion>
```
Comment (ViktorHofer) [inline src/installer/tests/Assets/TestProjects/TestWindowsOsShimsApp/TestWindowsOsShimsApp.csproj:10]: You should not hardcode these values here. Instead you want to add this line to `TestProjects.targets` which is imported here and in other projects:

```
  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory),global.json))\Directory.Build.targets" />
```

File: src/tests/Common/scripts/run-corefx-tests.py [hunk 13]
```diff
@@ -297,7 +297,7 @@ def main(args):
 
     # Gather up some arguments to pass to the different build scripts.
 
-    common_config_args = '-configuration Release -framework net5.0 -os %s -arch %s' % (clr_os, arch)
+    common_config_args = '-configuration Release -framework net6.0 -os %s -arch %s' % (clr_os, arch)
```
Comment (ViktorHofer) [inline src/tests/Common/scripts/run-corefx-tests.py:300]: who uses this script? cc @trylek @jkoritzinsky

File: src/tests/Common/external/external.csproj [hunk 14]
```diff
@@ -12,9 +12,9 @@
          This RID value doesn't really matter, since the assets we are copying are not RID specific, so defaulting to Windows here
     -->
     <OutputPath>$(TargetingPackPath)</OutputPath>
-    <TargetFramework>$(NetCoreAppCurrent)</TargetFramework>
-    <NuGetTargetMoniker>$(NetCoreAppCurrentTargetFrameworkMoniker)</NuGetTargetMoniker>
-    <NuGetTargetMonikerShort>$(NetCoreAppCurrent)</NuGetTargetMonikerShort>
+    <TargetFramework>net5.0</TargetFramework>
+    <NuGetTargetMoniker>$(NetCoreAppCurrentIdentifier),Version=v5.0</NuGetTargetMoniker>
+    <NuGetTargetMonikerShort>net5.0</NuGetTargetMonikerShort>
```
Comment (ViktorHofer) [inline src/tests/Common/external/external.csproj:17]: Why is this necessary? Also why are `NuGetTargetMoniker` and `NuGetTargetMonikerShort` properties set? I would not expect them to have any meaning anymore.

File: Directory.Build.targets (hunk 9 above)
Comment (Anipik) [reply Directory.Build.targets:26]: https://github.com/dotnet/sdk/issues/13950

**Comment by Anipik** [comment]
@marek-safar @akoeplinger can you help with the browser errors here ? i m not aware how everything is put together.
From the error it seems  an bin log it seems like that ```$(CORE_ROOT)``` property is not getting set.

File: src/coreclr/tests/scripts/run-corefx-tests.py (hunk 6 above)
Comment (trylek) [reply src/coreclr/tests/scripts/run-corefx-tests.py]: No idea really, I suspect this is some old pre-consolidation junk.

**Comment by Anipik** [comment]
@ViktorHofer @ericstj @safern this one is complete, can you please take a look ?

File: src/coreclr/src/tools/aot/crossgen2/crossgen2.csproj [hunk 15]
```diff
@@ -3,7 +3,7 @@
     <AssemblyName>crossgen2</AssemblyName>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <OutputType>Exe</OutputType>
-    <TargetFramework>$(NetCoreAppCurrent)</TargetFramework>
+    <TargetFramework>net5.0</TargetFramework>
```
Comment (ViktorHofer) [inline src/coreclr/src/tools/aot/crossgen2/crossgen2.csproj:6]: why not `$(NetCoreAppToolCurrent)`?

File: src/mono/wasm/debugger/BrowserDebugHost/BrowserDebugHost.csproj [hunk 16]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk.Web">
 
   <PropertyGroup>
-    <TargetFramework>$(NetCoreAppCurrent)</TargetFramework>
+    <TargetFramework>net5.0</TargetFramework>
```
Comment (ViktorHofer) [inline src/mono/wasm/debugger/BrowserDebugHost/BrowserDebugHost.csproj:4]: Why not `$(NetCoreAppToolCurrent)`?

File: src/mono/wasm/debugger/BrowserDebugProxy/BrowserDebugProxy.csproj [hunk 17]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>$(NetCoreAppCurrent)</TargetFramework>
+    <TargetFramework>net5.0</TargetFramework>
```
Comment (ViktorHofer) [inline src/mono/wasm/debugger/BrowserDebugProxy/BrowserDebugProxy.csproj:4]: same here.

Comment (ViktorHofer) [reply src/mono/wasm/debugger/BrowserDebugProxy/BrowserDebugProxy.csproj:4]: and in other files below.

File: src/tests/Common/Coreclr.TestWrapper/Coreclr.TestWrapper.csproj [hunk 18]
```diff
@@ -2,8 +2,8 @@
 
   <PropertyGroup>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
-    <NuGetTargetMoniker>$(NetCoreAppCurrentTargetFrameworkMoniker)</NuGetTargetMoniker>
-    <NuGetTargetMonikerShort>$(NetCoreAppCurrent)</NuGetTargetMonikerShort>
+    <NuGetTargetMoniker>$(NetCoreAppCurrentToolTargetFrameworkMoniker)</NuGetTargetMoniker>
+    <NuGetTargetMonikerShort>$(NetCoreAppToolCurrent)</NuGetTargetMonikerShort>
```
Comment (ViktorHofer) [inline src/tests/Common/Coreclr.TestWrapper/Coreclr.TestWrapper.csproj:6]: Why are these properties needed? Don't we want to set a tfm here? cc @ericstj @trylek

File: src/tests/Common/wasm-test-runner/WasmTestRunner.proj [hunk 19]
```diff
@@ -7,6 +7,7 @@
     <TargetFramework>$(NetCoreAppCurrent)</TargetFramework>
     <BuildDir>$(MSBuildThisFileDirectory)\obj\$(Configuration)\wasm</BuildDir>
     <AppDir>$(TestBinDir)/WasmApp/</AppDir>
+    <NETCoreAppMaximumVersion>99.0</NETCoreAppMaximumVersion>
```
Comment (ViktorHofer) [inline src/tests/Common/wasm-test-runner/WasmTestRunner.proj:10]: Why do you need to set this here? Does this project not import the root's Directory.Build.targets which already sets this?

File: src/libraries/System.Net.Http.Json/src/System.Net.Http.Json.csproj [hunk 20]
```diff
@@ -21,7 +22,7 @@
     <Compile Include="System\Net\Http\Json\TranscodingReadStream.cs" />
     <Compile Include="System\Net\Http\Json\TranscodingWriteStream.cs" />
   </ItemGroup>
-  <ItemGroup Condition="'$(TargetFramework)' == '$(NetCoreAppCurrent)'">
+  <ItemGroup Condition="'$(TargetFramework)' == '$(NetCoreAppCurrent)' or '$(TargetFramework)' == 'net5.0'">
```
Comment (ViktorHofer) [inline src/libraries/System.Net.Http.Json/src/System.Net.Http.Json.csproj:25]: nit: consider using a negation here to avoid hardcoding the previous netcoreapp version: `'$(TargetFramework)' != 'netstandard2.0' and '$(TargetFramework)' != 'net461'.

File: eng/testing/linker/project.csproj.template [hunk 21]
```diff
@@ -6,6 +6,7 @@
     <RuntimeIdentifier>{RuntimeIdentifier}</RuntimeIdentifier>
     <RuntimePackDir>{RuntimePackDir}</RuntimePackDir>
     <TargetingPackDir>{TargetingPackDir}</TargetingPackDir>
+    <NETCoreAppMaximumVersion>99.0</NETCoreAppMaximumVersion>
```
Comment (ViktorHofer) [inline eng/testing/linker/project.csproj.template:9]:
```suggestion
    <NETCoreAppMaximumVersion>{NetCoreAppMaximumVersion}</NETCoreAppMaximumVersion>
```

Comment (ViktorHofer) [reply eng/testing/linker/project.csproj.template:9]: and here an entry for it: https://github.com/dotnet/runtime/blob/19f97bd4d4440d09a8ff6a3efa7b4c9ea1207190/eng/testing/linker/trimmingTests.targets#L76. We try to avoid the 99.0 setting and use the current version instead.

File: src/mono/wasm/debugger/BrowserDebugHost/BrowserDebugHost.csproj (hunk 16 above)
Comment (Anipik) [reply src/mono/wasm/debugger/BrowserDebugHost/BrowserDebugHost.csproj:4]: I didnt use ```$(NetCoreAppToolCurrent)``` bcoz these projects are not tools. The restriction  here to not able to use ```$(NetCoreAppCurrent)``` is that we dont have a local aspnetcore targetingpack with net6.0 paths.

File: src/tests/Common/wasm-test-runner/WasmTestRunner.proj (hunk 19 above)
Comment (Anipik) [reply src/tests/Common/wasm-test-runner/WasmTestRunner.proj:10]: This project runs in helix which is outside the repo, hence the common settings donot work

File: src/mono/wasm/debugger/BrowserDebugHost/BrowserDebugHost.csproj (hunk 16 above)
Comment (ViktorHofer) [reply src/mono/wasm/debugger/BrowserDebugHost/BrowserDebugHost.csproj:4]: Makes sense, then I would recommend to define an additional property like `AspNetCoreCurrent`.

File: src/tests/Common/wasm-test-runner/WasmTestRunner.proj (hunk 19 above)
Comment (ViktorHofer) [reply src/tests/Common/wasm-test-runner/WasmTestRunner.proj:10]: How do other properties like `NetCoreAppCurrent` or `TestBinDir` work then?

Comment (Anipik) [reply src/tests/Common/wasm-test-runner/WasmTestRunner.proj:10]: @safern  do u know why these properties work ?

File: src/libraries/System.Net.Http.Json/ref/System.Net.Http.Json.csproj [hunk 22]
```diff
@@ -1,6 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>$(NetCoreAppCurrent);netstandard2.0;net461</TargetFrameworks>
+    <TargetFrameworks>$(NetCoreAppCurrent);net5.0;netstandard2.0;net461</TargetFrameworks>
```
Comment (ViktorHofer) [inline src/libraries/System.Net.Http.Json/ref/System.Net.Http.Json.csproj:3]: Curious, why are we adding a `net5.0` configuration to this package?

**Review by ViktorHofer — APPROVED** [review]
Cool 👍👍

File: Directory.Build.targets [hunk 23]
```diff
@@ -22,4 +22,27 @@
     <NETCoreAppMaximumVersion>$(MajorVersion).$(MinorVersion)</NETCoreAppMaximumVersion>
   </PropertyGroup>
 
+  <!--
+    SDK currently doesnot know about net6.0 TargetFramework.
```
Comment (safern) [inline Directory.Build.targets:26]:
```suggestion
    SDK currently does not know about net6.0 TargetFramework.
```

File: eng/testing/linker/project.csproj.template [hunk 24]
```diff
@@ -43,4 +44,23 @@
                       GeneratePathProperty="true" />
   </ItemGroup>
 
+  <ItemGroup>
```
Comment (safern) [inline eng/testing/linker/project.csproj.template:47]: Nit, these could be added into `supportFiles/Directory.Build.targets`

**Comment by ViktorHofer** [comment]
@Anipik before merging, please validate if building and testing inside Visual Studio still works. Also please validate that `dotnet test` on the CLI still works (we don't protect that yet in CI).

File: src/libraries/System.Security.Permissions/ref/System.Security.Permissions.csproj [hunk 25]
```diff
@@ -1,6 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>$(NetCoreAppCurrent);netstandard2.0;net461;netcoreapp3.0</TargetFrameworks>
+    <TargetFrameworks>$(NetCoreAppCurrent);net5.0;netstandard2.0;net461;netcoreapp3.0</TargetFrameworks>
```
Comment (safern) [inline src/libraries/System.Security.Permissions/ref/System.Security.Permissions.csproj:3]: Why does the ref need `NetCoreAppCurrent`?

File: src/libraries/pkg/test/project.csproj.template [hunk 26]
```diff
@@ -7,13 +7,27 @@
...
 
     <SupportFilesDir Condition="'$(SupportFilesDir)' == ''">$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), Directory.Build.props))\support</SupportFilesDir>
   </PropertyGroup>
-  
+
   <ItemGroup>
     <PackageReference Include="{PackageId}" Version="{PackageVersion}" />
   </ItemGroup>
 
+  <ItemGroup>
```
Comment (safern) [inline src/libraries/pkg/test/project.csproj.template:19]: Move this to `frameworkSettings/net6.0` as we also need to use the `RuntimeFrameworkVersion` property to test against latest `Microsoft.NETCore.App`.

File: src/libraries/pkg/test/project.csproj.template [hunk 27]
```diff
@@ -7,13 +7,27 @@
...
+    <KnownFrameworkReference Include="Microsoft.NETCore.App"
+      DefaultRuntimeFrameworkVersion="5.0.0-rc.1.20451.14"
+      IsTrimmable="true"
+      LatestRuntimeFrameworkVersion="5.0.0-rc.1.20451.14"
+      RuntimeFrameworkName="Microsoft.NETCore.App"
+      RuntimePackNamePatterns="Microsoft.NETCore.App.Runtime.**RID**"
+      RuntimePackRuntimeIdentifiers="linux-arm;linux-arm64;linux-musl-arm64;linux-musl-x64;linux-x64;osx-x64;rhel.6-x64;tizen.4.0.0-armel;tizen.5.0.0-armel;win-arm;win-arm64;win-x64;win-x86;ios-arm64;ios-arm;ios-x64;ios-x86;tvos-arm64;tvos-x64;android-arm64;android-arm;android-x64;android-x86;browser-wasm"
+      TargetFramework="net6.0"
+      TargetingPackName="Microsoft.NETCore.App.Ref"
+      TargetingPackVersion="5.0.0-rc.1.20451.14"/>
```
Comment (safern) [inline src/libraries/pkg/test/project.csproj.template:29]: Don't hardcode these versions and once you move this to `frameworkSettings/net6.0` use `RuntimeFrameworkVersion`.

File: src/libraries/pkg/test/project.csproj.template [hunk 28]
```diff
@@ -7,13 +7,27 @@
     <TestPackageVersion>{PackageVersion}</TestPackageVersion>
     <!-- Turn off end of life target framework checks as we intentionally build older .NETCoreApp configurations. -->
     <CheckEolTargetFramework>false</CheckEolTargetFramework>
+    <NETCoreAppMaximumVersion>99.0</NETCoreAppMaximumVersion>
```
Comment (safern) [inline src/libraries/pkg/test/project.csproj.template:10]: Nit, you could move this to: https://github.com/dotnet/runtime/blob/master/src/libraries/pkg/test/props/Directory.Build.props

File: src/libraries/shims/ApiCompatBaseline.PreviousNetCoreApp.txt [hunk 29]
```diff
@@ -1,170 +1,16 @@
-Compat issues with assembly netstandard:
```
Comment (safern) [inline src/libraries/shims/ApiCompatBaseline.PreviousNetCoreApp.txt:1]: Hopefully we don't need this anymore.

File: src/tests/Common/wasm-test-runner/WasmTestRunner.proj (hunk 19 above)
Comment (safern) [reply src/tests/Common/wasm-test-runner/WasmTestRunner.proj:10]: They are passed down globally: https://github.com/dotnet/runtime/blob/736b3042aa84de431838daf433de60dbaa645b4f/src/tests/Common/CLRTest.Execute.Bash.targets#L270

File: src/libraries/pkg/test/project.csproj.template (hunk 28 above)
Comment (Anipik) [reply src/libraries/pkg/test/project.csproj.template:10]: This property needs to be set in .csproj or in .targets otherwise it will be overwritten by the sdk.

File: src/libraries/shims/ApiCompatBaseline.PreviousNetCoreApp.txt (hunk 29 above)
Comment (Anipik) [reply src/libraries/shims/ApiCompatBaseline.PreviousNetCoreApp.txt:1]: we have couple of different apis

**Comment by Anipik** [comment]
> @Anipik before merging, please validate if building and testing inside Visual Studio still works. Also please validate that dotnet test on the CLI still works (we don't protect that yet in CI).


1.  Package sizes are same before and after the change
2. Dotnet test works from the CLI
3. Able to build projects form VS (we still get duplicate tfm error while restoring from the VS)
4. We are able to run tests using the VS

File: src/libraries/shims/ApiCompatBaseline.PreviousNetCoreApp.txt (hunk 29 above)
Comment (safern) [reply src/libraries/shims/ApiCompatBaseline.PreviousNetCoreApp.txt:1]: I meant the errors that you're removing 😄

**Review by safern — APPROVED** [review]

**Comment by Anipik** [comment]
Merging the change as failure is known and we have a tracking issue.

**Comment by ViktorHofer** [comment]
Note that this is a breaking change which ideally would go trough a monthly rollout. Please send out a mail to the team explaining why this was merged now and what the breaking changes is about.
